glide-core/telemetry/src/open_telemetry.rs

```diff
@@ -73,6 +73,41 @@ pub enum GlideOpenTelemetrySignalsExporter {
     File(PathBuf),
 }
 
+/// Signal types supported when reading protocol configuration from the
+/// environment.
+#[derive(Clone, Copy)]
+enum OtelSignal {
+    Traces,
+    Metrics,
+}
+
+/// Parse the OTLP protocol environment variables for the given signal. Returns
+/// `Some(Protocol)` when the environment specifies a supported protocol,
+/// otherwise `None`.
+fn protocol_from_env(signal: OtelSignal) -> Option<Protocol> {
+    fn parse(value: &str) -> Option<Protocol> {
+        match value.to_ascii_lowercase().as_str() {
+            "grpc" => Some(Protocol::Grpc),
+            "http/protobuf" | "http/binary" | "http" | "http_proto" => Some(Protocol::HttpBinary),
+            "http/json" => Some(Protocol::HttpJson),
+            _ => None,
+        }
+    }
+
+    let specific = match signal {
+        OtelSignal::Traces => "OTEL_EXPORTER_OTLP_TRACES_PROTOCOL",
+        OtelSignal::Metrics => "OTEL_EXPORTER_OTLP_METRICS_PROTOCOL",
+    };
+
+    if let Ok(val) = std::env::var(specific) {
+        parse(&val)
+    } else if let Ok(val) = std::env::var("OTEL_EXPORTER_OTLP_PROTOCOL") {
```

Comment:
so if the user passes both specific and general, we will silently ignore the general one. Is it the desired behavior? if so, please extend the documentation of this function:
```
/// Parse the OTLP protocol environment variables for the given signal.
/// Priority:
/// 1. OTEL_EXPORTER_OTLP_{TRACES|METRICS}_PROTOCOL
/// 2. OTEL_EXPORTER_OTLP_PROTOCOL (fallback)
/// 
/// Returns `Some(Protocol)` if the env var is set to a supported value, otherwise `None`.
```